@dcl/mappings.ts

```diff
@@ -77,3 +78,21 @@ function getProgressPayload(task: TaskState) {
       };
   }
 }
+
+function gatherRewards(tasks: any[]): RewardForRenderer[] {
+  const rewards: RewardForRenderer[] = [];
+
+  for (const task of tasks) {
+    for (const reward of task?.rewards) {
+      rewards.push({
+        id: reward.id ?? "",
+        name: reward.name ?? "",
+        type: reward.type ?? "",
+        imageUrl: "",
+        status: task?.givenRewards.find((x: any) => x.reward == reward).status ?? "ok",
```

Comment:
```suggestion
        status: task?.givenRewards.find((x: any) => x.reward == reward)?.status ?? "not_given",
```

You may need to signal somehow that the reward wasn't yet given to the player, if no corresponding given reward has been found

Reply:
Yeah, sounds good! gonna commit it from here!